src/main/java/com/salesforce/mce/awesolog/aws/S3ClientConstructor.java

```diff
@@ -26,37 +32,48 @@ public class S3ClientConstructor {
 
     private String s3Region;
 
+    private String s3Endpoint;
+
     public S3ClientConstructor(
         String awsAccessKey,
         String awsSecretKey,
         String awsSessionToken,
         String awsAssumeRoleArn,
-        String s3Region
-    ) {
+        String s3Region,
+        String s3Endpoint
+        ) {
         this.awsAccessKey = awsAccessKey;
         this.awsSecretKey = awsSecretKey;
         this.awsSessionToken = awsSessionToken;
         this.awsAssumeRoleArn = awsAssumeRoleArn;
         this.s3Region = s3Region;
+        this.s3Endpoint = s3Endpoint;
     }
 
     public S3Client construct() {
 
-        S3Client s3Client;
+        S3Client s3Client = null;
         StsClient stsClient = null;
 
-        S3ClientBuilder s3ClientBuilder;
-        StsClientBuilder stsClientBuilder;
+        S3ClientBuilder s3ClientBuilder = S3Client.builder();
+        StsClientBuilder stsClientBuilder = StsClient.builder();
 
         StaticCredentialsProvider staticCredentialsProvider = null;
 
         if (s3Region != null) {
             Region region = Region.of(s3Region);
-            s3ClientBuilder = S3Client.builder().region(region);
-            stsClientBuilder = StsClient.builder().region(region);
+            s3ClientBuilder = s3ClientBuilder.region(region).forcePathStyle(true);
+            stsClientBuilder = stsClientBuilder.region(region);
         } else {
-            s3ClientBuilder = S3Client.builder();
-            stsClientBuilder = StsClient.builder();
+            Region region = Region.US_WEST_2;
```

Comment:
do we need this? shouldn't the client infer region with default client setting, instead of hard coding it here?

Reply:
fixed